app/src/main/java/ch/epfl/sweng/swenggolf/offer/ListAnswerAdapter.java

```diff
@@ -185,9 +180,25 @@ public void onClick(DialogInterface dialog, int which) {
         return builder.create();
     }
 
-    private void writeFavPos(int pos) {
+    private void writeFavPos(final int pos) {
+        final int previousFavorite = answers.getFavoritePos();
         answers.setFavoritePos(pos);
-        Database.getInstance().write(Database.ANSWERS_PATH, offer.getUuid(), answers);
+        Database.getInstance().write(Database.ANSWERS_PATH, offer.getUuid(), answers,
+                new CompletionListener() {
+                    @Override
+                    public void onComplete(DbError error) {
+                        if (error == DbError.NONE) {
+                            if (previousFavorite != -1) {
+                                DatabaseUser.addPointsToUserId(-RESPOND_OFFER.getValue(),
+                                        answers.getAnswerList().get(previousFavorite).getUserId());
+                            }
+                            if (pos != -1) {
```

Comment:
The -1 means that there is no favorite user. Thus I might remove points to the previous favorite and add points to another user or I might just remove points or just add them. I cannot use an else if there, since you cannot have previous and pos being equal, this would mean that the user reselected the same favorite. The -1 wasn't a constant when I used it so kept using it the same way, I'll put a constant. If I didn't get your remark please tell me.

Reply:
Thank you for revealing the  absence of readability of the code.